src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const listOfNewStates = [];
+
+  const oneAction = (task) => {
+    switch (task.type) {
+      case 'addProperties':
+        Object.assign(newState, task.extraData);
+        break;
+      case 'removeProperties':
+        task.keysToRemove.forEach((key) => delete newState[key]);
+        break;
+      case 'clear':
+        Object.keys(newState).forEach((key) => delete newState[key]);
+    }
```

Comment:
The switch statement should include a default case to handle any unexpected 'task.type' values.